<launch>


    <arg name="use_pid" default="false"/>
    <arg name="use_pure_pursuit" default="false"/>
    <arg name="use_Stanley" default="false"/>
    <arg name="use_pure_pid" default="true"/>
    <!-- 启动仿真环境 -->
    <include file="$(find gazebo_pkg)/launch/race.launch"/>    

    <!-- 设置地图的配置文件 -->
    <arg name="map" default="mymap.yaml" />

    <!-- 运行地图服务器，并且加载设置的地图-->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find gazebo_map)/map/$(arg map)"/>

    <!-- 运行move_base节点 -->
    <include file="$(find my_navigation)/launch/teb_base.launch"/>

    <!-- 启动AMCL节点 -->
    <include file="$(find my_navigation)/launch/amcl.launch" />
    <!-- 显示odom节点 -->
    <node name="showodom" pkg="my_navigation" type="showpath" />
    <!-- 显示localPath节点 -->
    <node name="kdtree" pkg="my_navigation" type="kdtree" />	
    <!-- pid pursuit -->
    <group if="$(arg use_pid)">
    <node name="my_control" pkg="my_navigation" type="my_control" respawn="false" output="screen">
	<param name="g_fP" value="10"/>
	<param name="g_fD" value="5"/>
        <param name="Lfc" value="0.5"/>
        <param name="linear" value="1.5"/>
    </node>
   </group>
    <!-- pure pursuit -->
    <group if="$(arg use_pure_pursuit)">
    <node name="pure_pursuit" pkg="my_navigation" type="pure_pursuit" respawn="false" output="screen">
	 <param name="Lfc" value="0.5"/> 
         <param name="linear" value="2"/>
    </node>
    </group>
 <!-- Stanley pursuit -->
    <group if="$(arg use_Stanley)">
    <node name="Stanley" pkg="my_navigation" type="Stanley" respawn="false" output="screen">
	 <param name="Lfc" value="0.5"/> 
         <param name="linear" value="2"/>
    </node>
    </group>
 <!-- pure pid pursuit -->
    <group if="$(arg use_pure_pid)">
    <node name="pure_pid_pursuit" pkg="my_navigation" type="pure_pid_pursuit" respawn="false" output="screen">
	 <param name="Lfc" value="0.5"/> 
         <param name="linear" value="1.5"/>
         <param name="Kp" value="1.3"/>
         <param name="Kd" value="1"/>
    </node>
    </group>
    <!-- 运行rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find my_navigation)/rviz/teb_nav.rviz"/>
 
</launch>
